cmd/kubeadm/app/cmd/alpha/certs.go

```diff
@@ -170,36 +169,21 @@ func newCmdGenCSR() *cobra.Command {
 	return cmd
 }
 
-// runGenCSR contains the logic of the generate-csr sub-command after all
-// command line flags have been processed.
+// runGenCSR contains the logic of the generate-csr sub-command.
 func runGenCSR(config *genCSRConfig) error {
-	certDir := config.kubeadmConfig.CertificatesDir
-	certTree, err := certsphase.GetDefaultCertList().AsMap().CertTree()
+	leafCertificates, err := certsphase.LeafCertificates(certsphase.GetDefaultCertList())
```

Comment:
non-blocking nit (no need to amend). ideally the refactors should have been before the addition of the new logic.